pkg/suggestion/nasrl_service.py

```diff
@@ -1,22 +1,227 @@
-import random
-import string
-
+from pkg.suggestion.NAS_Reinforcement_Learning.Controller import Controller
+from pkg.suggestion.NAS_Reinforcement_Learning.Operation import SearchSpace
+from pkg.suggestion.NAS_Reinforcement_Learning.SuggestionParam import parseSuggestionParam
+import tensorflow as tf
 import grpc
-import numpy as np
-
 from pkg.api.python import api_pb2
 from pkg.api.python import api_pb2_grpc
 import logging
 from logging import getLogger, StreamHandler, INFO, DEBUG
+import json
+import os
 
 
 class NasrlService(api_pb2_grpc.SuggestionServicer):
-    def __init__(self, logger=None):
+    def __init__(self):
         self.manager_addr = "vizier-core"
         self.manager_port = 6789
+        self.current_study_id = ""
         self.current_trial_id = ""
+        self.ctrl_cache_file = ""
+        self.ctrl_step = 0
+        self.tf_graph = tf.get_default_graph()
+        self.is_first_run = True
+
+    def reset_controller(self, request):
+        print("-" * 80 + "\nResetting Suggestion for StudyJob {}\n".format(request.study_id) + "-" * 80)
+        self.ctrl_step = 0
+        self.current_study_id = request.study_id
+        self.ctrl_cache_file = "ctrl_cache/{}.ckpt".format(self.current_study_id)
```

Comment:
please store checkpoint of a controller by pvc in case of suggestion-naslr service pod restarts sowehow.

Reply:
Sure, will add it in TODO. I will add the support for multiple jobs, multiple trials, change to use tf package and change to use PVC within one month after this PR is merged.